shared/AppInsightsCore/Tests/Selenium/CorePerfCheck.Tests.ts

```diff
@@ -758,7 +757,7 @@ export class CorePerfCheckTests extends AITestClass {
                     let devAllowance = (theDuration.deviation / 2);
                     if (retryCount >= 5 || ((theDuration.duration + devAllowance) <= maxTime && (theDuration.iteration < maxIteration || (theDuration.avgIteration - devAllowance) < maxIteration))) {
                         // Last Retry or Good run
-                        if (!baseMeasurements || theDuration.deviation < baseMeasurements.deviation * 3) {
+                        if (retryCount >= 5 || !baseMeasurements || theDuration.deviation < baseMeasurements.deviation * 2.5) {
```

Comment:
By missing this check it caused the timeout issues from above because it just kept running attempts -- eventually timing out after 1 hour -- way too long. If it's bad failing the tests faster.

Reply:
Probably also the reason for the unhandled rejection promise issue...